csharp/src/Google.Protobuf.Test.NativeAot/Reflection/Descriptors.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System.Collections;
+using ProtobufTestMessages.Proto3;
+using UnitTest.Issues.TestProtos;
+
+namespace Google.Protobuf.Test.NativeAot.Reflection
+{
+    internal static class Descriptors
+    {
+        public static void GetFileDescriptorMessages()
+        {
+            var messageTypes = TestMessagesProto3Reflection.Descriptor.MessageTypes;
+            Assert.AreNotEqual(0, messageTypes.Count);
+            
+            foreach (var item in messageTypes)
+            {
+                Console.WriteLine($"{item.Name} message");
+            }
+            Console.WriteLine($"{messageTypes.Count} messages");
+        }
+        
+        public static void GetMessageFields()
+        {
+            var fields = TestAllTypesProto3.Descriptor.Fields.InDeclarationOrder();
+            Assert.AreNotEqual(0, fields.Count);
+            
+            foreach (var item in fields)
+            {
+                Console.WriteLine($"{item.Name} field");
+            }
+            Console.WriteLine($"{fields.Count} fields");
+        }
+
+        public static void GetMessageEnumTypes()
+        {
+            var enumTypes = TestAllTypesProto3.Descriptor.EnumTypes;
+            Assert.AreNotEqual(0, enumTypes.Count);
+
+            foreach (var item in enumTypes)
+            {
+                Console.WriteLine($"{item.Name} enum type");
+            }
+            Console.WriteLine($"{enumTypes.Count} enum types");
+        }
+
+        public static void GetEnumFields()
+        {
+            // There isn't a descriptor for enum types. Use standard .NET reflection to check fields.
```

Comment:
Yes there is - there's EnumDescriptor. Use `MessageDescriptor.EnumTypes` in this case to fetch it.
(Checking the actual enum is present with all fields as well probably isn't a bad idea too...)

Reply:
Immediately above this is a method that uses enum descriptors 😄